rfc-167-auto-patch-dependencies.md

```diff
@@ -0,0 +1,234 @@
+
+## Implementation plan
+
+The Platform teams will take the following actions to enable this RFC:
+
+### 1. Change GitHub repository configuration
+
+Since the [GOV.UK convention on GitHub Actions changed](https://github.com/alphagov/govuk-developer-docs/pull/4340) from a single `test` "job" to a wider `CI` "workflow" (made up of multiple jobs), it's harder to programmatically enforce branch protection. There are [branch protection rules configured in govuk-saas-config](https://github.com/alphagov/govuk-saas-config/blob/23fc44244488be135a6074a7c8b798227401452d/github/lib/configure_repo.rb#L16), but this entirely [relies on each CI job being explicitly listed in an overrides file](https://github.com/alphagov/govuk-saas-config/blob/23fc44244488be135a6074a7c8b798227401452d/github/repo_overrides.yml#L321-L326), so there is significant risk that this will fall out of sync with the latest list of job names. There is additional complexity around [some checks being legitimately 'skipped'](https://github.com/alphagov/support-api/actions/runs/7293511321/job/19876707509), which should not block the auto-merge.
+
+The Platform teams will reconsider how to automate branch protection, as part of a general move to [retire govuk-saas-config](https://trello.com/c/mojlsebq/3074-kill-off-govuk-saas-config). There are a few different approaches we could take, which need not be decided in this RFC, but suffice to say that we commit to ensuring that security scans are automatically set as required/blocking checks. We hope to make it as easy as practically possible for teams to adhere to the constraints outlined in this RFC, but the onus will still be on teams to self-certify whether or not a given repo or dependency update should be eligible for auto-merging.
+
+Note that this is an additional layer of defence. govuk-dependabot-merger [already checks that the `CI` workflow exists and all its jobs have succeeded](https://github.com/alphagov/govuk-dependabot-merger/pull/30), and will not attempt to approve and merge PRs that do not satisfy this condition. So, in theory, it should already be impossible for Dependabot PRs to be merged if a test suite or security scan fails.
+
+### 2. Change govuk-dependabot-merger
+
+The Platform teams will create a configuration option for govuk-dependabot-merger, allowing downstream repos to 'opt in' to external dependency patching. The current "internal only" behaviour will continue to be a supported option.
+
+The config for an opted in repository currently looks like:
+
+```yaml
+api_version: 1
+auto_merge:
+  - dependency: govuk_publishing_components
+    allowed_semver_bumps:
+      - patch
+      - minor
+  - dependency: rubocop-govuk
+    allowed_semver_bumps:
+      - patch
+      - minor
+```
+
+This RFC proposes a breaking change to the API as follows:
+
+```yaml
+api_version: 2
+default_behaviour:
+  external_dependencies_allowed: true # default: `false`
+  allow_automerging: true # default: `false`
```

Comment:
Just `auto_merge`? Since we're configuring the thing that's responsible for the auto-merging, there's really no "allowing" going on; it's either told to auto-merge or not.

(Same for the key under `dependency` ofc.)